frontend/src/lib/lemon-ui/LemonRadio/LemonRadio.tsx

```diff
@@ -0,0 +1,46 @@
+import clsx from 'clsx'
+
+import { IconCheckmark, IconRadioButtonUnchecked } from '../icons'
+
+export interface LemonRadioOption<T extends React.Key> {
+    label: string | JSX.Element
+    value: T
+}
+
+export interface LemonRadioProps<T extends React.Key> {
+    value?: T
+    onChange: (newValue: T) => void
+    options: LemonRadioOption<T>[]
+    disabled?: boolean
+    fullWidth?: boolean
+}
+
+/** Single choice radio. */
+export function LemonRadio<T extends React.Key>({
+    value,
+    onChange,
+    options,
+    disabled,
+    fullWidth,
+}: LemonRadioProps<T>): JSX.Element {
+    return (
+        <div
+            className={clsx('flex flex-col gap-2', fullWidth && 'w-full', disabled && 'text-muted cursor-not-allowed')}
+        >
+            {options.map((option) => (
+                <div
+                    key={option.value}
+                    className={clsx('flex items-center space-x-1', !disabled && 'cursor-pointer')}
+                    onClick={() => !disabled && onChange(option.value)}
+                >
+                    {option.value === value ? (
+                        <IconCheckmark className={clsx('text-lg', !disabled && 'text-primary-3000')} />
```

Comment:
It's really really key that a radio button has a circle in the middle, and not a checkmark. Checkmarks are for checkboxes. It's such a [classic and widely understood UI pattern](https://tonsky.me/blog/checkbox/) that it isn't a great idea to break out